tasks/post-bigquery-metrics-task.yaml

```diff
@@ -0,0 +1,122 @@
+
+    Obviously, such an approach with this task has limitations that we can't capture the status of the overall pipeline
+    including the `finally` tasks and that it does not provide visibility into the failures before or after the
+    pipeline.
+    However, it still may be useful for the situation when the majority of the observed failures is in the pipeline
+    itself.
+  params:
+  - name: AGGREGATE_TASKS_STATUS
+    default: ''
+    description: |
+      Status of normal (non-final) tasks of the pipeline.
+      When this task is at the beginning of the pipeline, don't pass anything.
+      When the task is placed in the pipeline's finally block, pass here `$(tasks.status)`.
+      Ref https://tekton.dev/docs/pipelines/pipelines/#using-aggregate-execution-status-of-all-tasks
+  volumes:
+  - name: gcp-service-account
+    secret:
+      optional: false
+      secretName: konflux-metrics-gcp-service-account
+  steps:
+  - name: update-bigquery
+    # GCP CLIs aren't available in Quay or in the Red Hat Registry. If the availability of Google's registry becomes a
+    # problem for our pipelines stability, we can set up image mirroring.
+    image: gcr.io/google.com/cloudsdktool/google-cloud-cli:stable@sha256:767516bf8e50f22c074c07e1138844744b80345e5ca315ea413a88ab22e36089
+    volumeMounts:
+    - name: gcp-service-account
+      mountPath: /mnt/gcp-service-account
+    env:
+    - name: PIPELINE_RUN_NAME
+      # E.g. "scanner-v4-db-on-push-477w8"
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.labels['tekton.dev/pipelineRun']
+    - name: ORIGINAL_PIPELINE_RUN_NAME
+      # E.g. "scanner-v4-db-on-push"
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.labels['pipelinesascode.tekton.dev/original-prname']
+    - name: REPO_URL
+      # E.g. "https://github.com/stackrox/stackrox"
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.annotations['pipelinesascode.tekton.dev/repo-url']
+    - name: SOURCE_BRANCH
+      # E.g. "konflux/references/master"
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.annotations['pipelinesascode.tekton.dev/source-branch']
+    - name: PULL_REQUEST_NUMBER
+      # E.g. "14577"
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.labels['pipelinesascode.tekton.dev/pull-request']
+    - name: COMMIT_SHA
+      # E.g. "2bb555465e2b822abb4eb13da2167908670ff5c9"
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.labels['pipelinesascode.tekton.dev/sha']
+    script: |
+      #!/usr/bin/env bash
+
+      set -euo pipefail
+
+      GCP_PROJECT="acs-san-stackroxci"
+      TABLE_NAME="${GCP_PROJECT}.ci_metrics.stackrox_jobs"
+
+      gcloud auth activate-service-account --key-file /mnt/gcp-service-account/konflux-metrics-gcp-service-account
+      gcloud config set project "${GCP_PROJECT}"
+
+      AGGREGATE_TASKS_STATUS="$(params.AGGREGATE_TASKS_STATUS)"
```

Comment:
I mirror your feelings about stopped pipeline runs getting the `outcome: Completed`, but I see no other way as to record this as is and have our interpretation. The semantic of `Completed` might change in a future Tekton Pipelines version, which would make a lookup take mapping Completed to Cancelled incorrect.
 
Can you update the column description in https://console.cloud.google.com/bigquery?referrer=search&inv=1&invt=AbtJ8Q&project=acs-san-stackroxci&ws=!1m5!1m4!4m3!1sacs-san-stackroxci!2sci_metrics!3sstackrox_jobs

> Succeeded or Completed / Failed / None (likely means cancelled).
to 
> Succeeded or Completed (likely stopped) / Failed (cancelled or failed) / None (running or timed out in `finally`).

Reply:
Oh yes, will update the description thanks for reminding.

Reply:
Updated to
> For prow: passed / failed / canceled. For GHA: success / failure / cancelled. For Konflux: Succeeded / Failed (including timed out and cancelled) / Completed (manually stopped - not a success) / None (not seen yet). NULL means either the job is still running or it was aborted before it was able to report the final status.